### PR TITLE
intermountainneuroimaing/hcp-icafix:1.06.15_inc1.1

### PR DESCRIPTION
**Pull request by amyhegarty**
Flywheel gear wrapping hcp-icafix. Accepts flywheel gear bids-hcp outputs as gear arguments. Current functionality limited to single run denoising.

### CODE REVIEW

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  "url": "https://github.com/Washington-University/Pipelines",
+  "source": "https://github.com/intermountainneuroimaging/hcp-icafix",
+  "cite": "(1) G. Salimi-Khorshidi, G. Douaud, C.F. Beckmann, M.F. Glasser, L. Griffanti S.M. Smith. Automatic denoising of functional MRI data: Combining independent component analysis and hierarchical fusion of classifiers. NeuroImage, 90:449-68, 2014 (2) L. Griffanti, G. Salimi-Khorshidi, C.F. Beckmann, E.J. Auerbach, G. Douaud, C.E. Sexton, E. Zsoldos, K. Ebmeier, N. Filippini, C.E. Mackay, S. Moeller, J.G. Xu, E. Yacoub, G. Baselli, K. Ugurbil, K.L. Miller, and S.M. Smith. ICA-based artefact removal and accelerated fMRI acquisition for improved resting state network imaging. NeuroImage, 95:232-47, 2014",
+  "version": "1.06.15_inc1.1",
+  "custom": {
+    "gear-builder": {
+      "image": "amhe4269/hcp-icafix:1.06.15_inc1.1",
+      "category": "analysis"
+    },
+      "flywheel": {
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:17]:
```suggestion
    "flywheel": {
```

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      "suite": "Denoising"
+    }
+  },
+  "config": {
+     "gear-log-level": {
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:22]:
```suggestion
    "gear-log-level": {
```

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      "description": "Gear Log verbosity level (ERROR|WARNING|INFO|DEBUG)",
+      "type": "string",
+      "enum": [
+        "ERROR",
+        "WARNING",
+        "INFO",
+        "DEBUG"
+      ]
+    },
+
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:33]:
```suggestion
```

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      "default": "HCP_hp2000.RData",
+      "enum": ["HCP_hp2000.RData",
+              "HCP7T_hp2000.RData",
+              "HCP_Style_Single_Multirun_Dedrift.RData",
+              "NHPHCP_Macaque_RIKEN30SRFIX.RData",
+              "Standard.RData",
+              "Standard_old.Rdata",
+              "UKBiobank.RData",
+              "WhII_MB6.RData",
+              "WhII_Standard.RData"],
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:45]:
```suggestion
      "enum": [
        "HCP_hp2000.RData",
        "HCP7T_hp2000.RData",
        "HCP_Style_Single_Multirun_Dedrift.RData",
        "NHPHCP_Macaque_RIKEN30SRFIX.RData",
        "Standard.RData",
        "Standard_old.Rdata",
        "UKBiobank.RData",
        "WhII_MB6.RData",
        "WhII_Standard.RData"
      ],
```

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+      "description": "Name of FIX training file to use for classification. 'Standard', 'HCP_hp2000', 'HCP7T_hp2000', 'WhII_MB6', 'WhII_Standard', 'UKBiobank'. See https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FIX/UserGuide#Trained-weights_files for details."
+    },
+
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:48]:
```suggestion
```

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    "DeleteIntermediates": {
+      "type": "boolean",
+      "default": false,
+      "description": "delete highpass files (note that delete intermediates=TRUE is not recommended for MR+FIX)"
+    },
+    "dry-run": {
+        "type":"boolean",
+        "default": false,
+        "description": "Log all commands, but do not execute."
+    }
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:75]:
```suggestion
    "dry-run": {
      "type":"boolean",
      "default": false,
      "description": "Log all commands, but do not execute."
    }
```

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  },
+  "inputs": {
+    "api-key": {
+        "base": "api-key"
+    },
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:80]:
```suggestion
    "api-key": {
      "base": "api-key"
    },
```

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    "functional_zip": {
+      "description": "Select functional run for ICA",
+      "base": "file",
+      "optional": false
+    },
+
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:86]:
```suggestion
```

File: gears/intermountainneuroimaging/hcp-icafix.json [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    "structural_zip": {
+      "description": "Select structural zip for the functional data",
+      "base": "file",
+      "optional": false
+    }
+
```
Comment (pvelasco) [inline gears/intermountainneuroimaging/hcp-icafix.json:92]:
```suggestion
```

**Review by pvelasco — COMMENTED** [review]
Thank you very much for updating this gear for the Exchange!

I just had a few minor formatting suggestions. After addressing those, I'll pull your changes.

**Comment by amyhegarty** [comment]
Hi Pablo, thanks for the suggestions. I approved all the suggested changes. Thanks!

**Comment by pvelasco** [comment]
Thank you for the contribution, @amyhegarty !

**Pull request merged**